vsintegration/src/FSharp.Editor/LanguageService/SingleFileWorkspaceMap.fs

```diff
@@ -74,7 +82,45 @@ type internal SingleFileWorkspaceMap(workspace: VisualStudioWorkspace,
 
         member _.OnAfterFirstDocumentLock(_, _, _, _) = VSConstants.E_NOTIMPL
 
-        member _.OnAfterSave(_) = VSConstants.E_NOTIMPL
+        member _.OnAfterSave(docCookie) = 
+            // This causes re-analysis to happen when a F# document is saved.
+            // We do this because FCS relies on the file system and existing open documents
+            // need to be re-analyzed so the changes are propogated.
+            // We only re-analyze F# documents that are dependent on the document that was just saved.
+            // We ignore F# script documents here.
+            let vsTextBuffer = rdt4.GetDocumentData(docCookie) :?> IVsTextBuffer
+            let textBuffer = editorAdaptersFactory.GetDataBuffer(vsTextBuffer)
+
+            let textContainer = textBuffer.AsTextContainer()
+            let mutable workspace = Unchecked.defaultof<_>
+            if Workspace.TryGetWorkspace(textContainer, &workspace) then
+                let solution = workspace.CurrentSolution
+                let documentId = workspace.GetDocumentIdInCurrentContext(textContainer)
```

Comment:
Is this just open documents, or all documents?

Reply:
Just open documents